packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -75,15 +76,22 @@ class CupertinoPageTransition extends AnimatedWidget {
     // TODO(ianh): tell the transform to be un-transformed for hit testing
     // but not while being controlled by a gesture.
     return new SlideTransition(
-      position: _outgoingPositionAnimation,
+      position: _secondaryPositionAnimation,
       child: new SlideTransition(
-        position: _incomingPositionAnimation,
-        child: new PhysicalModel(
-          shape: BoxShape.rectangle,
-          color: _kBackgroundColor,
-          elevation: 32,
+        position: _primaryPositionAnimation,
+        child: new DecoratedBox(
+          decoration: new BoxDecoration(
+            boxShadow: <BoxShadow>[
```

Comment:
this shadow will look weird, since it's not a material shadow. Is that what we want?

In any case: BoxDecoration itself knows how to lerp(). I would hard-code two constant BoxDecorations, and then lerp between them by passing the lerp function the appropriate value from 0.0 to 1.0 based on the animation.

We could even abstract that out into a ContainerTransition similar to AnimatedContainer.